src/fs_main.ts

```diff
@@ -0,0 +1,889 @@
+
+  try {
+    const response: axios.AxiosResponse = await axios.post(API_URL, data, {
+      headers
+    })
+
+    if (response.status === 200) {
+      const responseData = response.data
+
+      if (responseData.errors) {
+        throw new Error(`Error: ${responseData.errors}`)
+      }
+
+      return responseData
+    } else {
+      throw new Error(`Error: ${response.status} - ${response.statusText}`)
+    }
+  } catch (error) {
+    if (isAxiosError(error)) {
+      throw new Error(
+        `Error: ${error.response?.status} - ${error.response?.statusText}`
+      )
+    } else {
+      throw error
+    }
+  }
+}
+
+async function createTestAsThirdPartyScanner(
+  token: string,
+  organizationContext: string,
+  params: {
+    businessUnitId: string
+    createdByUserId: string
+    assetId: string
+    artifactId: string
+    testName: string
+    testType: string
+    uploadMethod: UploadMethod
+    productId?: string
+  }
+): Promise<CreateTestResponse> {
+  const {
+    businessUnitId,
+    createdByUserId,
+    assetId,
+    artifactId,
+    testName,
+    testType,
+    uploadMethod = UploadMethod.API,
+    productId
+  } = params
+  return createTest(token, organizationContext, {
+    businessUnitId,
+    createdByUserId,
+    assetId,
+    artifactId,
+    testName,
+    testType,
+    uploadMethod,
+    productId
+  })
+}
+interface CreateTestResponse {
+  createTest: {
+    id: string
+    name: string
+    artifactUnderTest: {
+      id: string
+      name: string
+      assetVersion: {
+        id: string
+        name: string
+        asset: {
+          id: string
+          name: string
+          dependentProducts: {
+            id: string
+            name: string
+          }[]
+        }
+      }
+    }
+    createdBy: {
+      id: string
+      email: string
+    }
+    ctx: {
+      asset: string
+      products: string[]
+      businessUnits: string[]
+    }
+    uploadMethod: string
+  }
+}
+
+async function createTest(
+  token: string,
+  organizationContext: string,
+  params: {
+    businessUnitId: string
+    createdByUserId: string
+    assetId: string
+    artifactId: string
+    testName: string
+    testType: string
+    tools?: any[]
+    uploadMethod?: UploadMethod
+    productId?: string
+  }
+): Promise<CreateTestResponse> {
+  const {
+    businessUnitId,
+    createdByUserId,
+    assetId,
+    artifactId,
+    testName,
+    testType,
+    tools = [],
+    uploadMethod = UploadMethod.API,
+    productId
+  } = params
+  if (!businessUnitId) {
+    throw new Error('Business unit ID is required')
+  }
+  if (!createdByUserId) {
+    throw new Error('Created by user ID is required')
+  }
+  if (!assetId) {
+    throw new Error('Asset ID is required')
+  }
+  if (!artifactId) {
+    throw new Error('Artifact ID is required')
+  }
+  if (!testName) {
+    throw new Error('Test name is required')
+  }
+  if (!testType) {
+    throw new Error('Test type is required')
+  }
+
+  const graphqlQuery = `
```

Comment:
Do we want to move all of the queries and mutations in here to `./fs_queries.ts`?

Reply:
> Do we want to move all of the queries and mutations in here to `./fs_queries.ts`?

Do you mean move it to another file?

Reply:
Nope not at all!